packages/@aws-cdk/aws-codedeploy/lib/deployment-group.ts

```diff
@@ -121,6 +132,25 @@ export interface ServerDeploymentGroupProps {
      * @default ServerDeploymentConfig#OneAtATime
      */
     deploymentConfig?: IServerDeploymentConfig;
+
+    /**
+     * The auto-scaling groups belonging to this Deployment Group.
+     *
+     * @default []
+     */
+    autoScalingGroups?: autoscaling.AutoScalingGroup[];
+
+    /**
+     * If you've provided any auto-scaling groups with the {@link #autoScalingGroups} property,
+     * you can set this property to automatically install the CodeDeploy agent on the instances
+     * using User Data.
+     * The value specifies what family does the operating system of the auto-scaling groups
+     * belongs to, as the installation varies between different systems.
+     *
+     * @default the agent will not be automatically installed
+     * @see https://docs.aws.amazon.com/codedeploy/latest/userguide/codedeploy-agent-operations-install.html
+     */
+    agentInstallationType?: AgentInstallationType;
```

Comment:
Isn't it possible to deduce the OS type from the AutoScalingGroup?

Reply:
Yes, except the instructions are different for different flavors of Linux.

However, after consulting with the CodeDeploy team, I've decided to try and write one script that will work on all of the Linux distributions. Will change.